org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/preferences/OptionsConfigurationBlock.java

```diff
@@ -528,7 +531,7 @@ private void incrementRebuildCount() {
 	}
 
 	protected boolean processChanges(IWorkbenchPreferenceContainer container) {
-		boolean needsBuild = !getPreferenceChanges().isEmpty();
+		boolean needsBuild = buildPreferenceEvaluator.isAffectingBuild(getPreferenceChanges());

```

Comment:
why is buider here in a generic class

Reply:
maybe a simple method here and the delegate to the evaluator in subclass

Reply:
@szarnekow requested an injectable API for better customization

Reply:
This is so much entangled, it's a shame, eg see this code below `container.registerUpdateJob(getBuildJob(getProject()));`

Bindings required for the builder are currently done by the Xtext generator, which I'd like to avoid.

It would be a safe bet if we moved the XtextBuildPreferenceEvaluator and defined the constant for the output config in ui and referenced it from `EclipseOutputConfigurationProvider.OUTPUT_PREFERENCE_TAG`